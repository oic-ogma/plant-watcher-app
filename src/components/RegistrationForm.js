import React, { Component } from 'react';
import { Error, Label, FormInputComponent, HideKeyboardOnPress } from './common';
import { Button } from 'react-native-elements';

export default class RegistrationForm extends Component {
  onButtonPress () {
    const { email, password, registerUser } = this.props;
    registerUser(email, password);
  }

  renderButton () {

    return (
      <Button onPress={this.onButtonPress.bind(this)}
        raised
        large
        loading={this.props.loading}
        loadingRight={true}
        title='登録'
        backgroundColor='#228b22'/>

    );
  }

  render() {
    const {
      email,
      password,
      emailChanged,
      passwordChanged,
      error
    } = this.props;

    return (
      <HideKeyboardOnPress>
        <Label placeholder='メールアドレス'/>
        <FormInputComponent
          placeholder='example@hoge.com'
          onChangeText={emailChanged}
          value={email}
        />

        <Label placeholder='パスワード'/>
        <FormInputComponent
          secureTextEntry
          placeholder='パスワード'
          onChangeText={passwordChanged}
          value={password}
        />

        <Error message={error}/>

        {this.renderButton()}

      </HideKeyboardOnPress>
    );
  }
}
