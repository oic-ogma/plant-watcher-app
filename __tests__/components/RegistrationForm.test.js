import React from 'react';
import RegistrationForm from '../../src/components/RegistrationForm';
import renderer from 'react-test-renderer';

describe('登録フォーム', () => {
  it('snapshot-普通', () => {
    const tree = renderer.create(<RegistrationForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-メールアドレスとパスワードが入力されている', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: 'PassWord',
      error: '',
      loading: false
    };
    const tree = renderer.create(<RegistrationForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ロード中', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: 'PassWord',
      error: '',
      loading: true
    };
    const tree = renderer.create(<RegistrationForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-エラーが出ている', () => {
    const INITIAL_STATE = {
      email: 'example@gmail.com',
      password: '',
      error: 'Error',
      loading: true
    };
    const tree = renderer.create(<RegistrationForm {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
