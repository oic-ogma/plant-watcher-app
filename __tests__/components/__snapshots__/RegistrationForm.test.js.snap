// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`登録フォーム snapshot-エラーが出ている 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        メールアドレス
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="example@hoge.com"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="example@gmail.com"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        パスワード
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="パスワード"
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    Error
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;

exports[`登録フォーム snapshot-メールアドレスとパスワードが入力されている 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        メールアドレス
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="example@hoge.com"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="example@gmail.com"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        パスワード
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="パスワード"
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="PassWord"
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 18,
            "fontWeight": "600",
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        登録
      </Text>
    </View>
  </View>
</View>
`;

exports[`登録フォーム snapshot-ロード中 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        メールアドレス
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="example@hoge.com"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="example@gmail.com"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        パスワード
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="パスワード"
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="PassWord"
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;

exports[`登録フォーム snapshot-普通 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        メールアドレス
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="example@hoge.com"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        パスワード
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={undefined}
        placeholder="パスワード"
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 18,
            "fontWeight": "600",
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        登録
      </Text>
    </View>
  </View>
</View>
`;
